### PR TITLE
fix: replace mock tests with real tests in http logger

### PR DESCRIPTION
**Pull request by Revolyssup**
### Description

Replaces existing mock tests in http logger with real tests using vector as http log server
### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by Sn0rt — APPROVED** [review]

File: t/plugin/http-logger-log-format.t [hunk 1]
```diff
@@ -438,7 +439,7 @@ done
                 [[{
                     "log_format": {
                         "host": "$host",
-                        "labels": "$a6_route_labels",
+                        "labels": "custom variable check",
```
Comment (monkeyDluffy6017) [inline t/plugin/http-logger-log-format.t:442]: Why you change this?

File: t/plugin/udp-logger.t [hunk 2]
```diff
@@ -425,7 +425,6 @@ qr/.*plugin_metadata.*/
             end
 
             ngx.say(body)
-
```
Comment (monkeyDluffy6017) [inline t/plugin/udp-logger.t:428]: The `ngx.say` has nothing to do with the code below, so the blank is needed?

Comment (Revolyssup) [reply t/plugin/udp-logger.t:428]: Yes but I am unsure. I was getting lint error previously so I added this new line.

File: t/plugin/http-logger-log-format.t (hunk 1 above)
Comment (Revolyssup) [reply t/plugin/http-logger-log-format.t:442]: Actually the labels passed in route were not being reflected on the log so I had removed $a6_route_labels. Checking this right now...

Comment (Revolyssup) [reply t/plugin/http-logger-log-format.t:442]: Okay this ctx variable was set here https://github.com/apache/apisix/blob/73ac77d22b4ae0c2b597105e778f9a5d6ea86e3d/t/plugin/http-logger-log-format.t#L491
Added it back

**Comment by monkeyDluffy6017** [comment]
Could you merge the master? The ci always failed because the go packages faile to download.

**Comment by monkeyDluffy6017** [comment]
@Revolyssup please resolve the conflicts

**Comment by Revolyssup** [comment]
> @Revolyssup please resolve the conflicts

@monkeyDluffy6017 Done

**Comment by Revolyssup** [comment]
@monkeyDluffy6017 Please approve CI

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**